File: aspnet-core/src/Lesson.Application/Domain/ClassRoom/Dto/UpdateClassRoomInput.cs
using Abp.Domain.Entities;
using Lesson.Authorization.Users;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Lesson.Domain.ClassRoom.Dto
{
    public class UpdateClassRoomInput:Entity<int>
    {
        public string Name { get; set; }
        public string Branch { get; set; }
        public string Description { get; set; }
        public List<Lesson.Dto.LessonFullOutPut> Lessons { get; set; }
        public List<UserPartOutPut> Students { get; set; }


    }
}

File: angular/src/app/services/ClassRoom/dtos/UpdateClassRoomInput.ts
import { UserPartOutPut } from './../../User/dtos/UserPartOutPut';
import { LessonFullOutPut } from './../../Lesson/dtos/LessonFullOutPut';
export class UpdateClassRoomInput implements IUpdateClassRoomInput {
    id: number;
    name: string;
    branch: string;
    description: string;
    lessons: LessonFullOutPut[];
    students:UserPartOutPut[];

}

export interface IUpdateClassRoomInput{
    id:number;
    name:string|undefined;
    branch:string|undefined;
    description:string|undefined;
    lessons:LessonFullOutPut[]|undefined;
    students:UserPartOutPut[]|undefined;
}
